### ISSUE TITLE
Ensure categories are decoded and displayed properly when added from workspace settings from newDot

### ISSUE THREAD

**Issue by MonilBhavsar** [Expensify/App#36473]
### Problem

Context: https://expensify.slack.com/archives/C03TQ48KC/p1707467669800489
Categories added from workspace settings in oldDot are stored in encoded format and we need to decode at the client side to display it properly in decoded format
 

### Why it is important

To display and store categories properly and consistently across all layers of the App.
We have this bug in newDot where categories added in encoded format are not displayed correctly https://github.com/Expensify/App/issues/34735 and we decided not to decode for categories added from oldDot.

### Solution

When adding/importing categories from workspace settings in newDot, ensure category name is decoded and displayed properly throughout the application

**Comment by MonilBhavsar** [comment]
cc @luacmartins since you're working on wave8 simplified collect categories project

**Comment by MonilBhavsar** [comment]
@sonialiap this will be handled internally, feel free to unassign 😄

**Comment by FitseTLT** [comment]
@MonilBhavsar take a look at https://github.com/Expensify/App/issues/36622 it was created and closed yesterday it is related to ampersand inconsitently getting html encoded and decoded on category leading to a crash I reproduced it before I wrote my [proposal](https://github.com/Expensify/App/issues/36622#issuecomment-1947005762) but it was closed by `Bug` member due to inconsistency of reproduction of the issue.

**Comment by MonilBhavsar** [comment]
Putting it on HOLD

**Comment by MonilBhavsar** [comment]
Still on HOLD

**Comment by FitseTLT** [comment]
@MonilBhavsar have U checked https://github.com/Expensify/App/issues/36622#issuecomment-1973349785

**Comment by trjExpensify** [comment]
What are the next steps for this issue, @MonilBhavsar? If we put something on hold, let's reference the specific issue it's held on. A lot of the categories issues/PRs have merged, so it's unclear if this is unblocked. Thanks! CC: @luacmartins

**Comment by MonilBhavsar** [comment]
Held on https://github.com/Expensify/App/issues/35710 as we can start adding new categories from newDot after that issue.
> What are the next steps for this issue

After the held issue is closed. We need to ensure we are decoding and storing categories in our DB. Most probably will need backend change for that

**Comment by trjExpensify** [comment]
Thanks for making the dependency clear! @luacmartins has a PR in review for the CreateCategoriesPage, so shouldn't be too long a hold now!

**Comment by MonilBhavsar** [comment]
Held issue is closed, taking it off hold and I'll take a look

**Comment by trjExpensify** [comment]
@MonilBhavsar where you at on this one? Is this issue still necessary?

**Comment by MonilBhavsar** [comment]
I'll get back to this early next week. This can be treated as polish item for the wave-collect

**Comment by MonilBhavsar** [comment]
Got back here and working on this!

**Comment by MonilBhavsar** [comment]
Dealing with some track tax issues, circling back

**Comment by MonilBhavsar** [comment]
Moving it to daily and wrapping it up!

**Comment by MonilBhavsar** [comment]
Discussing about the solution here https://expensify.slack.com/archives/C03TQ48KC/p1714480744709159?thread_ts=1707467669.800489&cid=C03TQ48KC

**Comment by MonilBhavsar** [comment]
We decided to do nothing 
> Whatever the user typed is what we should store in the DB and the same thing is what they should see in the app (which is not the same thing as what we should output in HTML

If user enters category in encoded format, we store and display it in encode format.
